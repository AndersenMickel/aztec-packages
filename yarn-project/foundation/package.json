{
  "name": "@aztec/foundation",
  "version": "0.1.0",
  "packageManager": "yarn@3.4.1",
  "type": "module",
  "main": "./dest/index.js",
  "types": "./dest/index.d.ts",
  "exports": {
    "./eslint": "./.eslintrc.cjs",
    "./eslint.docs": "./.eslintrc.docs.cjs",
    "./prettier": "./.prettierrc.json",
    "./abi": {
      "bun": "./src/abi/index.ts",
      "default": "./dest/abi/index.js"
    },
    "./async-map": {
      "bun": "./src/async-map/index.ts",
      "default": "./dest/async-map/index.js"
    },
    "./aztec-address": {
      "bun": "./src/aztec-address/index.ts",
      "default": "./dest/aztec-address/index.js"
    },
    "./collection": {
      "bun": "./src/collection/index.ts",
      "default": "./dest/collection/index.js"
    },
    "./crypto": {
      "bun": "./src/crypto/index.ts",
      "default": "./dest/crypto/index.js"
    },
    "./error": {
      "bun": "./src/error/index.ts",
      "default": "./dest/error/index.js"
    },
    "./eth-address": {
      "bun": "./src/eth-address/index.ts",
      "default": "./dest/eth-address/index.js"
    },
    "./fifo": {
      "bun": "./src/fifo/index.ts",
      "default": "./dest/fifo/index.js"
    },
    "./json-rpc": {
      "bun": "./src/json-rpc/index.ts",
      "default": "./dest/json-rpc/index.js"
    },
    "./json-rpc/server": {
      "bun": "./src/json-rpc/server/index.ts",
      "default": "./dest/json-rpc/server/index.js"
    },
    "./json-rpc/client": {
      "bun": "./src/json-rpc/client/index.ts",
      "default": "./dest/json-rpc/client/index.js"
    },
    "./log": {
      "bun": "./src/log/index.ts",
      "default": "./dest/log/index.js"
    },
    "./mutex": {
      "bun": "./src/mutex/index.ts",
      "default": "./dest/mutex/index.js"
    },
    "./fields": {
      "bun": "./src/fields/index.ts",
      "default": "./dest/fields/index.js"
    },
    "./retry": {
      "bun": "./src/retry/index.ts",
      "default": "./dest/retry/index.js"
    },
    "./running-promise": {
      "bun": "./src/running-promise/index.ts",
      "default": "./dest/running-promise/index.js"
    },
    "./serialize": {
      "bun": "./src/serialize/index.ts",
      "default": "./dest/serialize/index.js"
    },
    "./sleep": {
      "bun": "./src/sleep/index.ts",
      "default": "./dest/sleep/index.js"
    },
    "./timer": {
      "bun": "./src/timer/index.ts",
      "default": "./dest/timer/index.js"
    },
    "./transport": {
      "bun": "./src/transport/index.ts",
      "default": "./dest/transport/index.js"
    },
    "./trees": {
      "bun": "./src/trees/index.ts",
      "default": "./dest/trees/index.js"
    },
    "./wasm": {
      "bun": "./src/wasm/index.ts",
      "default": "./dest/wasm/index.js"
    },
    "./worker": {
      "bun": "./src/worker/index.ts",
      "default": "./dest/worker/index.js"
    },
    "./bigint-buffer": {
      "bun": "./src/bigint-buffer/index.ts",
      "default": "./dest/bigint-buffer/index.js"
    },
    "./types": {
      "bun": "./src/types/index.ts",
      "default": "./dest/types/index.js"
    },
    "./url": {
      "bun": "./src/url/index.ts",
      "default": "./dest/url/index.js"
    },
    "./committable": {
      "bun": "./src/committable/index.ts",
      "default": "./dest/committable/index.js"
    },
    "./noir": {
      "bun": "./src/noir/index.ts",
      "default": "./dest/noir/index.js"
    },
    "./testing": {
      "bun": "./src/testing/index.ts",
      "default": "./dest/testing/index.js"
    },
    "./array": {
      "bun": "./src/array/index.ts",
      "default": "./dest/array/index.js"
    },
    "./validation": {
      "bun": "./src/validation/index.ts",
      "default": "./dest/validation/index.js"
    }
  },
  "scripts": {
    "build": "yarn clean && tsc -b",
    "build:dev": "tsc -b --watch",
    "clean": "rm -rf ./dest .tsbuildinfo",
    "formatting": "run -T prettier --check ./src && run -T eslint ./src",
    "formatting:fix": "run -T eslint --fix ./src && run -T prettier -w ./src",
    "test:light": "NODE_NO_WARNINGS=1 node --experimental-vm-modules $(yarn bin jest) --passWithNoTests",
    "test": "NODE_NO_WARNINGS=1 node --experimental-vm-modules $(yarn bin jest) --passWithNoTests"
  },
  "inherits": [
    "../package.common.json"
  ],
  "jest": {
    "preset": "ts-jest/presets/default-esm",
    "moduleNameMapper": {
      "^(\\.{1,2}/.*)\\.[cm]?js$": "$1"
    },
    "testRegex": "./src/.*\\.test\\.(js|mjs|ts)$",
    "rootDir": "./src"
  },
  "dependencies": {
    "@aztec/bb.js": "portal:../../barretenberg/ts",
    "@koa/cors": "^4.0.0",
    "@noble/curves": "^1.2.0",
    "bn.js": "^5.2.1",
    "debug": "^4.3.4",
    "detect-node": "^2.1.0",
    "elliptic": "^6.5.4",
    "hash.js": "^1.1.7",
    "koa": "^2.14.2",
    "koa-bodyparser": "^4.4.0",
    "koa-compress": "^5.1.0",
    "koa-router": "^12.0.0",
    "leveldown": "^6.1.1",
    "levelup": "^5.1.1",
    "lodash.chunk": "^4.2.0",
    "lodash.clonedeepwith": "^4.5.0",
    "memdown": "^6.1.1",
    "pako": "^2.1.0",
    "sha3": "^2.1.4",
    "ts-essentials": "^9.4.1",
    "zod": "^3.22.4"
  },
  "devDependencies": {
    "@jest/globals": "^29.5.0",
    "@types/bn.js": "^5.1.3",
    "@types/debug": "^4.1.7",
    "@types/detect-node": "^2.0.0",
    "@types/elliptic": "^6.4.16",
    "@types/jest": "^29.5.0",
    "@types/koa": "^2.13.5",
    "@types/koa-bodyparser": "^4.3.10",
    "@types/koa-compress": "^4.0.3",
    "@types/koa-cors": "^0.0.2",
    "@types/koa-router": "^7.4.4",
    "@types/koa__cors": "^4.0.0",
    "@types/leveldown": "^4.0.3",
    "@types/levelup": "^5.1.2",
    "@types/lodash.chunk": "^4.2.9",
    "@types/lodash.clonedeepwith": "^4.5.7",
    "@types/memdown": "^3.0.1",
    "@types/node": "^18.7.23",
    "@types/pako": "^2.0.0",
    "@types/supertest": "^2.0.12",
    "@typescript-eslint/eslint-plugin": "^6.2.1",
    "@typescript-eslint/parser": "^6.2.1",
    "comlink": "^4.4.1",
    "eslint": "^8.21.0",
    "eslint-config-prettier": "^8.5.0",
    "eslint-plugin-jsdoc": "^40.1.0",
    "eslint-plugin-no-only-tests": "^3.1.0",
    "eslint-plugin-tsdoc": "^0.2.17",
    "jest": "^29.5.0",
    "prettier": "^2.7.1",
    "supertest": "^6.3.3",
    "ts-jest": "^29.1.0",
    "ts-node": "^10.9.1",
    "typescript": "^5.0.4"
  },
  "files": [
    "dest",
    "src",
    "!*.test.*"
  ],
  "engines": {
    "node": ">=18"
  }
}
