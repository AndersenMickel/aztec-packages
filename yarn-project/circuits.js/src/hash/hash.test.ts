import { times } from '@aztec/foundation/collection';

import { AztecAddress, Fr, SideEffect, SideEffectLinkedToNoteHash } from '../index.js';
import { makeAztecAddress, makeVerificationKey } from '../tests/factories.js';
import {
  computeCommitmentNonce,
  computeCommitmentsHash,
  computeMessageSecretHash,
  computeNullifierHash,
  computePublicDataTreeLeafSlot,
  computePublicDataTreeValue,
  computeUniqueCommitment,
  computeVarArgsHash,
  hashVK,
  siloNoteHash,
  siloNullifier,
} from './hash.js';

describe('hash', () => {
  it('hashes VK', () => {
    const vk = makeVerificationKey();
    const res = hashVK(vk.toBuffer());
    expect(res.toString('hex')).toMatchSnapshot();
  });

  it('computes commitment nonce', () => {
    const nullifierZero = new Fr(123n);
    const commitmentIndex = 456;
    const res = computeCommitmentNonce(nullifierZero, commitmentIndex);
    expect(res.toString()).toMatchSnapshot();
  });

  it('computes unique commitment', () => {
    const nonce = new Fr(123n);
    const innerCommitment = new Fr(456);
    const res = computeUniqueCommitment(nonce, innerCommitment);
    expect(res.toString()).toMatchSnapshot();
  });

  it('computes siloed commitment', () => {
    const contractAddress = new AztecAddress(new Fr(123n).toBuffer());
    const uniqueCommitment = new Fr(456);
    const res = siloNoteHash(contractAddress, uniqueCommitment);
    expect(res.toString()).toMatchSnapshot();
  });

  it('computes siloed nullifier', () => {
    const contractAddress = new AztecAddress(new Fr(123n).toBuffer());
    const innerNullifier = new Fr(456);
    const res = siloNullifier(contractAddress, innerNullifier);
    expect(res.toString()).toMatchSnapshot();
  });

  it('computes public data tree value', () => {
    const value = new Fr(3n);
    const res = computePublicDataTreeValue(value);
    expect(res.toString()).toMatchSnapshot();
  });

  it('computes public data tree leaf slot', () => {
    const contractAddress = makeAztecAddress();
    const value = new Fr(3n);
    const res = computePublicDataTreeLeafSlot(contractAddress, value);
    expect(res.toString()).toMatchSnapshot();
  });

  it('hashes empty function args', () => {
    const res = computeVarArgsHash([]);
    expect(res.toString()).toMatchSnapshot();
  });

  it('hashes function args', () => {
    const args = times(8, i => new Fr(i));
    const res = computeVarArgsHash(args);
    expect(res.toString()).toMatchSnapshot();
  });

  it('hashes many function args', () => {
    const args = times(200, i => new Fr(i));
    const res = computeVarArgsHash(args);
    expect(res.toString()).toMatchSnapshot();
  });

  it('compute secret message hash', () => {
    const value = new Fr(8n);
    const hash = computeMessageSecretHash(value);
    expect(hash.toString()).toMatchSnapshot();
  });

  it('Computes an empty nullifier hash ', () => {
    const emptyNull = SideEffectLinkedToNoteHash.empty();

    const emptyHash = computeNullifierHash(emptyNull).toString();
    expect(emptyHash.toString()).toMatchSnapshot();
  });

  it('Computes an empty sideeffect hash ', () => {
    const emptySideEffect = SideEffect.empty();
    const emptyHash = computeCommitmentsHash(emptySideEffect).toString();
    expect(emptyHash.toString()).toMatchSnapshot();
  });

  it('Var args hash matches noir', () => {
    const args = times(800, i => new Fr(i));
    const res = computeVarArgsHash(args);
    expect(res.toString()).toMatchSnapshot();

    // Value used in "compute_var_args_hash" test in hash.nr
    // console.log("hash", hash);
  });
});
