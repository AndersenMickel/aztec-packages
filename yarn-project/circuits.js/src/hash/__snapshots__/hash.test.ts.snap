// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hash Computes an empty nullifier hash  1`] = `"0x066e6cdc4a6ba5e4781deda650b0be6c12f975f064fc38df72c1060716759b17"`;

exports[`hash Computes an empty sideeffect hash  1`] = `"0x27b1d0839a5b23baf12a8d195b18ac288fcf401afb2f70b8a4b529ede5fa9fed"`;

exports[`hash Var args hash matches noir 1`] = `Fr<0x05a1023fef839ac88731f49ae983e172c1b600a3c8f3393ad0ac25d819ac0f0f>`;

exports[`hash compute secret message hash 1`] = `Fr<0x0dc06f2167e2cd19adf738d1f38469d7f8bff1e26b029816e8230bcd6ab6332e>`;

exports[`hash computes commitment nonce 1`] = `Fr<0x148c4666e2c5bce33dc53b30430808464f7ed76b28daf8a36722e93b49a31a5e>`;

exports[`hash computes public data tree leaf slot 1`] = `Fr<0x14114ab3dbdd0a1ccc5c4fe68dd576f3c6cd79708770e06ab4086398cdd828f4>`;

exports[`hash computes public data tree value 1`] = `Fr<0x0000000000000000000000000000000000000000000000000000000000000003>`;

exports[`hash computes siloed commitment 1`] = `Fr<0x148c4666e2c5bce33dc53b30430808464f7ed76b28daf8a36722e93b49a31a5e>`;

exports[`hash computes siloed nullifier 1`] = `Fr<0x148c4666e2c5bce33dc53b30430808464f7ed76b28daf8a36722e93b49a31a5e>`;

exports[`hash computes unique commitment 1`] = `Fr<0x148c4666e2c5bce33dc53b30430808464f7ed76b28daf8a36722e93b49a31a5e>`;

exports[`hash hashes VK 1`] = `Buffer<0x048110667f80b02f77b7d744976657cea9a7c5f1dd2340ea1c579a7ebfd54e55>`;

exports[`hash hashes empty function args 1`] = `Fr<0x0000000000000000000000000000000000000000000000000000000000000000>`;

exports[`hash hashes function args 1`] = `Fr<0x1a76e9750a1493d95ce48be1fa31831fc370d7e68f563fe5c781c6f58e1f1eac>`;

exports[`hash hashes many function args 1`] = `Fr<0x21e37f4da2762d4daff39c2337d812b9c7b9c46aa9fafdf666d7d7fd9935d124>`;
